cauldron/redis_cache.py

```diff
@@ -270,4 +278,240 @@ def apply_cache(*args, **kwargs):
+                key = self._get_key(namespace, key)
+            yield from redis.set(key, value, expire=expire)
+
+    @coroutine
+    def increment_value(self, key, namespace=None):
+        # Set a redis key and increment the value by one
+        with (yield from self.get_pool()) as redis:
+            if namespace is not None:
+                key = self._get_key(namespace, key)
+            yield from redis.incr(key)
+
+    @coroutine
+    def increment_by_value(self, key, value: int, namespace=None):
+        with (yield from self.get_pool()) as redis:
+            if namespace is not None:
+                key = self._get_key(namespace, key)
+            yield from redis.incrby(key, value)
+
+    @coroutine
+    def set_key_if_not_exists(self, key, value, namespace=None, expire=0):
+        """
+        Set a redis key and return True if the key does not exists else return False
+        :param key: Key name
+        :param value: Value
+        :param namespace : Namespace to associate the key with
+        :param expire: expiration
+        :return:
+        """
+        with (yield from self.get_pool()) as redis:
+            if namespace:
+                key = self._get_key(namespace, key)
+            return (yield from redis.set(key, value, expire=expire, exist='SET_IF_NOT_EXIST'))
+
+    @coroutine
+    def get_key(self, key, namespace=None):
+        with (yield from self.get_pool()) as redis:
+            if namespace is not None:
+                key = self._get_key(namespace, key)
+            return (yield from redis.get(key, encoding=self._utf8))
+
+    @coroutine
+    def hmget(self, fields, namespace=''):
+        with (yield from self.get_pool()) as redis:
+            return (yield from redis.hmget(namespace, *fields))
+
+    @coroutine
+    def hmset(self, field, value, namespace=''):
+        with (yield from self.get_pool()) as redis:
+            yield from redis.hmset(namespace, field, value)
+
+    @coroutine
+    def delete(self, key, namespace=None):
+        with (yield from self.get_pool()) as redis:
+            if namespace is not None:
+                key = self._get_key(namespace, key)
+            yield from redis.delete(key)
+
+    def hdel(self, key, namespace):
+        with (yield from self.get_pool()) as redis:
+            if namespace is not None:
+                yield from redis.hdel(namespace, key)
+
+    def hgetall(self, namespace):
+        with (yield from self.get_pool()) as redis:
+            if namespace is not None:
+                return (yield from redis.hgetall(namespace, encoding=self._utf8))
+
+    @coroutine
+    def clear_namespace(self, namespace) -> int:
+        pattern = namespace + '*'
+        return (yield from self._delete_by_pattern(pattern))
+
+    @coroutine
+    def _delete_by_pattern(self, pattern: str) -> int:
+        if not pattern:
+            return 0
+        with (yield from self.get_pool()) as redis:
+            _keys = yield from redis.keys(pattern)
```

Comment:
keys should be changed to scan, as it can degrade performance since this cache will be shared by many services.

if we are using redis version greater than 5 delete should be changed to unlink, unlink and scan does not block the redis server.

We should not expose keys method in the wrapper, it should not be used ... it would be better if we make delete_by_pattern method name to purge and make it private.  In any case one application can have one super namepsace and subsequent namespace for the managemet of keys, and purge can accept namespace to clear the keys..

Like for order_service:

global namespace -  order_service/1.0.0/
order_count_namespace - order_service/1.0.0/order_count